efficientdet/keras/efficientdet_keras.py

```diff
@@ -728,13 +728,27 @@ def __init__(self, model_name=None, config=None, name=''):
         strategy=config.strategy,
         data_format=config.data_format)
 
+    if self.config.moving_average_decay is not None:
+      self.ema = tf.train.ExponentialMovingAverage(
+          decay=self.config.moving_average_decay)
+      self.ema_vars = None
+
   def _init_set_name(self, name, zero_based=True):
     """A hack to allow empty model name for legacy checkpoint compitability."""
     if name == '':  # pylint: disable=g-explicit-bool-comparison
       self._name = name
     else:
       self._name = super().__init__(name, zero_based)
 
+  def build(self, input_shape):
+    super().build(input_shape)
+    ema_vars = self.trainable_variables
+    for v in self.variables:
+      # We maintain mva for batch norm moving mean and variance as well.
+      if 'moving_mean' in v.name or 'moving_variance' in v.name:
+        ema_vars.append(v)
+    self.ema_vars = ema_vars
```

Comment:
Where are ema vars created in eager mode?

Reply:
They will be created in train_step which is wrapped with tf.function.
The code only record variables that need to create ema.

Reply:
sg